crates/shared/src/price_estimation/native/oneinch.rs

```diff
@@ -0,0 +1,237 @@
+use {
+    super::{NativePriceEstimateResult, NativePriceEstimating},
+    crate::price_estimation::{
+        Estimate,
+        PriceEstimateResult,
+        PriceEstimating,
+        PriceEstimationError,
+        Query,
+    },
+    anyhow::{anyhow, Result},
+    futures::{future::BoxFuture, FutureExt},
+    model::order::OrderKind,
+    number::nonzero::U256 as NonZeroU256,
+    primitive_types::{H160, U256},
+    reqwest::Client,
+    std::{
+        collections::HashMap,
+        sync::{Arc, Mutex},
+        time::Duration,
+    },
+};
+
+const BASE_URL: &str = "https://api.1inch.dev/";
+const REFRESH_INTERVAL: Duration = Duration::from_secs(12);
```

Comment:
IIRC the long term plan is to synchronize these kinds of things with the `block_stream` module. That would allow us to only fetch new prices when we detect a new block. This has the potential to return outdated prices when 1Inch did not yet update their prices based on that block but so does a hardcoded interval.
Also hardcoding doesn't play nice with different chains (e.g. `GC` needs `5s`).